lib/yao/resources/user.rb

```diff
@@ -9,8 +9,11 @@ class User < Base
     self.resources_name = "users"
     self.admin          = true
 
-    def self.get_by_name(name)
-      self.list.find {|user| user.name == name }
+    class << self
+      def get_by_name(name)
+        self.get("", name: name)
```

Comment:
Now I found `self.list(name: name)` doesn't work for `User` unfortunately. because the key of the json response of `GET /users?name=foo` is `user`, not `users`.

IMO, this is a special case. Most of other APIs return plural results on querying. so I think introducing new flag is safe and simple way to fix this problem.

```ruby
module Yao::Resources
  class User < Base
    ...

    self.return_single_on_querying = true

    ...
  end
end
```

What do you think about it?